### ISSUE TITLE
freqUI displays incorrect percentage value

### ISSUE THREAD

**Issue by VerDoe** [freqtrade/freqtrade#10274]
* Operating system: Debian GNU/Linux 12 (bookworm)
  * Python Version: Python 3.11.2
  * CCXT version: ccxt==4.3.35
  * Freqtrade Version: freqtrade 2024.5
  
## Describe the problem:

The stoploss shows the wrong percentage value in the UI. However, the bot works with the correct percentage value. But the current stoploss dist is correct

![Screenshot 2024-05-30 225547](https://github.com/freqtrade/freqtrade/assets/155641519/0e0bfdbf-d00f-4e80-9ba2-16215252442a)
![Screenshot 2024-05-31 103602](https://github.com/freqtrade/freqtrade/assets/155641519/f6e3eeb7-ff41-417b-b5fb-5f3fa0e9d0b0)

**Comment by xmatthias** [comment]
What makes you think it's incorrect ? 

From the screenshot - i see no reason to belive that the value would be incorrect, other than a statement "it's incorrect".

**Comment by VerDoe** [comment]
In the first example: 0.0849 is not -2.188% of 0.0885. In the second example, the percentage value is also incorrect. However, the bot triggers the stoploss at the correct value. Also compare the other two percentages, which are correct. It shows the wrong percentage value.

**Comment by xmatthias** [comment]
well frequi only displays what's in the database - so i suspect that this is what's in the database.

Please provide the output of the following command for the affected trade:
if the values changed in the meantime, please use aligned screenshots.

`freqtrade show-trades --db-url sqlite:///<trades.sqlite> --print-json --trade-ids <tradeid>` 
Please use a "details block" to keep the issue itself kindof readable (see below for details on how to do that). 

<details>
<summary>Expand to see how to do details blocks</summary>

```
<details>
<summary>Expand full </summary>
(wrap in tripple ticks `)
Trades output ...
</details>
```
</details>

Now i doubt logs will help - as usually there's no log written about what a stoploss is set to unless you're running in debug mode - which you usually won't - as that's pretty verbose :laughing:

**Comment by VerDoe** [comment]
Here is another example. I only have a test bot running at the moment and delete everything from time to time. But I have noticed that it is not always the case.

![Screenshot 2024-05-31 210240](https://github.com/freqtrade/freqtrade/assets/155641519/ace2c4be-b796-43c8-bb2c-23be2dc29c14)
-----------------

```
[
    {
        "trade_id": 54,
        "pair": "WLD/USDT",
        "base_currency": "WLD",
        "quote_currency": "USDT",
        "is_open": false,
        "exchange": "binance",
        "amount": 20.2,
        "amount_requested": 20.26523197,
        "stake_amount": 99.0002,
        "max_stake_amount": 99.0002,
        "strategy": "Test",
        "enter_tag": "enter_0.786",
        "timeframe": 5,
        "fee_open": 0.001,
        "fee_open_cost": 0.09900020000000001,
        "fee_open_currency": "USDT",
        "fee_close": 0.001,
        "fee_close_cost": 0.095849,
        "fee_close_currency": "USDT",
        "open_date": "2024-05-31 12:20:29",
        "open_timestamp": 1717158029087,
        "open_fill_date": "2024-05-31 12:21:09",
        "open_fill_timestamp": 1717158069922,
        "open_rate": 4.901,
        "open_rate_requested": 4.901,
        "open_trade_value": 99.0992002,
        "close_date": "2024-05-31 15:59:29",
        "close_timestamp": 1717171169589,
        "realized_profit": -3.3460492,
        "realized_profit_ratio": -0.03376464384421944,
        "close_rate": 4.745,
        "close_rate_requested": 4.745,
        "close_profit": -0.03376464384421944,
        "close_profit_pct": -3.38,
        "close_profit_abs": -3.3460492,
        "trade_duration_s": 13140,
        "trade_duration": 219,
        "profit_ratio": -0.03376464384421944,
        "profit_pct": -3.38,
        "profit_abs": -3.3460492,
        "exit_reason": "trailing_stop_loss",
        "exit_order_status": "closed",
        "stop_loss_abs": 4.747,
        "stop_loss_ratio": -0.017346583850931574,
        "stop_loss_pct": -1.7346583850931574,
        "stoploss_last_update": null,
        "stoploss_last_update_timestamp": null,
        "initial_stop_loss_abs": 4.411,
        "initial_stop_loss_ratio": -0.1,
        "initial_stop_loss_pct": -10.0,
        "min_rate": 4.745,
        "max_rate": 4.97,
        "leverage": 1.0,
        "interest_rate": 0.0,
        "liquidation_price": null,
        "is_short": false,
        "trading_mode": "spot",
        "funding_fees": 0.0,
        "amount_precision": 1.0,
        "price_precision": 3.0,
        "precision_mode": 2,
        "contract_size": 1.0,
        "has_open_orders": false,
        "orders": [
            {
                "amount": 20.2,
                "safe_price": 4.901,
                "ft_order_side": "buy",
                "order_filled_timestamp": 1717158069922,
                "ft_is_entry": true,
                "ft_order_tag": "enter_0.786",
                "pair": "WLD/USDT",
                "order_id": "dry_run_buy_WLD/USDT_1717158028.812814",
                "status": "closed",
                "average": 4.901,
                "cost": 99.00019999999999,
                "filled": 20.2,
                "is_open": false,
                "order_date": "2024-05-31 12:20:28",
                "order_timestamp": 1717158028812,
                "order_filled_date": "2024-05-31 12:21:09",
                "order_type": "limit",
                "price": 4.901,
                "remaining": 0.0,
                "ft_fee_base": null,
                "funding_fee": null
            },
            {
                "amount": 20.2,
                "safe_price": 4.745,
                "ft_order_side": "sell",
                "order_filled_timestamp": 1717171169589,
                "ft_is_entry": false,
                "ft_order_tag": "trailing_stop_loss",
                "pair": "WLD/USDT",
                "order_id": "dry_run_sell_WLD/USDT_1717171169.305247",
                "status": "closed",
                "average": 4.745,
                "cost": 95.849,
                "filled": 20.2,
                "is_open": false,
                "order_date": "2024-05-31 15:59:29",
                "order_timestamp": 1717171169305,
                "order_filled_date": "2024-05-31 15:59:29",
                "order_type": "limit",
                "price": 4.745,
                "remaining": 0.0,
                "ft_fee_base": null,
                "funding_fee": 0.0
            }
        ]
    }
]
```

**Comment by xmatthias** [comment]
Does look correct to me (it's displaying what's in the trade object).

I can see why you think it's wrong though - but it's false expectation as to what this value represents, rather than the value it displays that's wrong.

Stoploss will be the value from open *until your value starts trailing* .

Once that's the case - it'll correspond to the value your `custom_stoploss()` function returned - hence representing the value below the current price - at the moment the stoploss has been updated.
This can be quite helpful in debugging [custom_stoploss](https://www.freqtrade.io/en/stable/strategy-callbacks/#custom-stoploss) functionalities and expectations - as this is what you're expected to return (a percentage of current price).

What this value doesn't represent (and is not trying to represent) is the relative stoploss against the open rate - at least not once the value starts trailing.
It represents this until the trade stops trailing it's stoploss - but that's accidental - as current_rate at that moment is the open-rate.

**Comment by VerDoe** [comment]
No, I didn't expect the relative stoploss either. I'll take the first screenshot again, as you can see it better there. If I take the current stoploos dist and calculate it with the current profit, I get exactly the percentage value of 0.0849 in percent (top right). This is a static value, namely the value of 0.0849 in percent. I would have expected the value next to the 0.0849 to be this value.
In other words, what does the percentage value next to 0.0849 represent?

**Comment by xmatthias** [comment]
well it's exactly what i said above - it's _what your strategy returned_ at the time the stoploss was updated.

taking your first screenshot - your current rate is 0.09. your current stoploss is at 0.0849 - the distance which current-stoploss-dist also shows in relative terms.
The absolute value next to "stoploss-dist" is essentially just that - `0.09 - 0.0849`.
The relative value next to current stoploss-dist displays exactly this - the relative distance between the current rate (0.9) - and the absolute stoploss (0.0849).

The relative value next to 0.0849 is the value your strategy returned when the stoploss was last updated.
the current rate at that point in time was essentially 0.0849 + 2.188%.

There's no "reference" value for that as such - as it's "current rate at a certain point in time".


Hope this helps

**Comment by VerDoe** [comment]
Thank you for the explanation!
